src/zenml/integrations/spark/step_operators/spark_step_operator.py

```diff
@@ -0,0 +1,325 @@
+                    "store, please make sure that your artifact store has"
+                    "defined the required credentials namely the access key "
+                    "and the secret access key."
+                )
+        else:
+            logger.warning(
+                "In most cases, the spark step operator requires additional "
+                "configuration based on the artifact store flavor you are "
+                "using. That also means, that when you use this step operator "
+                "with certain artifact store flavor, ZenML can take care of "
+                "the pre-configuration. However, the artifact store flavor "
+                f"'{artifact_store.FLAVOR}' featured in this stack is not "
+                f"known to this step operator and it might require additional "
+                f"configuration."
+            )
+
+    def _additional_configuration(self, spark_config: SparkConf) -> None:
+        """Appends the user-defined configuration parameters.
+
+        Args:
+            spark_config: a SparkConf object which collects all the
+                configuration parameters
+
+        Returns:
+            the same SparkConf object with appended configuration parameters
+        """
+        # Add the additional parameters
+        if self.submit_kwargs:
+            for k, v in self.submit_kwargs.items():
+                spark_config.set(k, v)
+
+    def _launch_spark_job(
+        self, spark_config: SparkConf, entrypoint_command: List[str]
+    ):
+        """Generates and executes a spark-submit command.
+
+        Args:
+            spark_config: a SparkConf object which collects all the
+                configuration parameters
+            entrypoint_command: a list of strings which is normally used to
+                execute the step as a module. However, in the context of this
+                step operator, this list is used to form the application
+                parameters for spark-submit.
+
+        Returns:
+            str, the complete spark-submit command
+        """
+        # Base spark-submit command
+        command = [
+            f"spark-submit "
+            f"--master {self.master} "
+            f"--deploy-mode {self.deploy_mode}"
+        ]
+
+        # Add the configuration parameters
+        command += [f"--conf {c[0]}={c[1]}" for c in spark_config.getAll()]
+
+        # Add the application path
+        command.append(self.application_path)
+
+        # Add the application parameters
+        for arg in [
+            "--main_module",
+            "--step_source_path",
+            "--execution_info_path",
+            "--input_artifact_types_path",
+        ]:
+            i = entrypoint_command.index(arg)
+            command.extend([arg, entrypoint_command[i + 1]])
+
+        command = " ".join(command)
+
+        # Execute the spark-submit
+        process = subprocess.Popen(
+            command,
```

Comment:
I think its not necessary to join all the command items into one single string, to the best of my knowledge Popen will also take a list of strings.

Reply:
You are absolutely right. I have kept it there for debugging purposes. I will update and resolve it.